tests/deploy_tests/test_updates.py

```diff
@@ -0,0 +1,730 @@
+        """
+        return self.get_location()["name"]
+
+    def get_version(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return self.get_location()["version"]
+
+    def _get_metadata(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return {
+            "frameworks": MockStore.instance.get_bundle(
+                self._type, self.get_system_name(), self.get_version()
+            ).required_frameworks
+        }
+
+    def run_post_install(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        pass
+
+    def exists_local(self):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        return True
+
+    @classmethod
+    def find_latest_item(cls, pc_path, bundle_install_path, bundle_type, name, constraint_pattern=None):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        if constraint_pattern:
+            return cls._find_latest_for_pattern(bundle_type, name, constraint_pattern)
+        else:
+            return cls._find_latest(bundle_type, name)
+
+    def find_latest_version(self, version_pattern=None):
+        """
+        See documentation from TankAppStoreDescriptor.
+        """
+        if version_pattern:
+            return self._find_latest_for_pattern(
+                self._type,
+                self.get_system_name(),
+                version_pattern
+            )
+        else:
+            return self._find_latest(
+                self._type,
+                self.get_system_name()
+            )
+
+    @classmethod
+    def _find_latest(cls, bundle_type, name):
+        """
+        Finds the latest version of a bundle.
+
+        :param bundle_type: Type of the bundle. Any of AppDescriptor.{APP,ENGINE,FRAMEWORK}
+        :param name: Name of the bundle.
+
+        :returns: A MockStore._Entry descriptor for the latest version of the bundle.
+        """
+
+        versions = MockStore.instance.get_bundle_versions(bundle_type, name)
+        latest = "v0.0.0"
+        for version in versions:
+            if LooseVersion(version) > LooseVersion(latest):
+                latest = version
+        return cls.create(name, latest, bundle_type)
+
+    @classmethod
+    def _find_latest_for_pattern(cls, bundle_type, name, version_pattern):
+        """
+        Finds the latest version of a bundle for a given version pattern, e.g. v1.x.x
+
+        :param bundle_type: Type of the bundle. Any of AppDescriptor.{APP,ENGINE,FRAMEWORK}
+        :param name: Name of the bundle.
+        :param version_pattern: Version pattern to use when searching for the latest version, e.g. v1.x.x.
+
+        :returns: A MockStore._Entry descriptor for the latest version of the bundle that matches the pattern.
+        """
+        # FIXME: Refactor this with the code from the new descriptor code
```

Comment:
Heads up that I think this may be pulled from https://github.com/shotgunsoftware/tk-core/blob/develop/0.18/python/tank_vendor/shotgun_deploy/io_descriptor/base.py#L110 in 0.18